src/PIL/JpegImagePlugin.py

```diff
@@ -205,7 +209,16 @@ def SOF(self, marker):
     if self.layers == 1:
         self.mode = "L"
     elif self.layers == 3:
-        self.mode = "RGB"
+        try:
+            transform_flag = self.info["adobe_transform"]
+            if transform_flag == 0:  # Unknown (RGB or CMYK)
+                self.mode = "YCbCr"
+            elif transform_flag in (1, 2):  # YCbCr or YCCK
+                self.mode = "RGB"
```

Comment:
```suggestion
                self.mode = "YCbCr"
```
Looks like you are setting these values the wrong way around, and that they are the correct way around in your comments.